### ISSUE TITLE
Validate required boolean field with `false`

### ISSUE THREAD

**Issue by moyuanhuang** [go-ozzo/ozzo-validation#184]
Hi, I noticed that when using the `Required` validator against a boolean field with value equals `false`, the validator will raise 

```
FIELD_NAME: cannot be blank.
```

Is this expected?

Sample code: https://go.dev/play/p/SkiEIx2aaQN

**Comment by rngallen** [comment]
Yes as per source code

 Required is a validation rule that checks if a value is not empty.
 A value is considered not empty if
- integer, float: not zero
- bool: true
- string, array, slice, map: len() > 0
- interface, pointer: not nil and the referenced value is not empty
- any other types

**Comment by slessard** [comment]
See this pull request in jellydator-validation that makes clear the behavior for booleans with the `Required` rule.

https://github.com/jellydator/validation/pull/18

**Comment by kolkov** [comment]
Yes, this is expected — `Required` treats the zero value of each type as "empty", and for `bool` that means `false`. It's documented behavior but admittedly unintuitive for booleans.

For a `*bool` field, use `validation.NotNil` instead of `Required` — that way both `true` and `false` are accepted, only `nil` (missing) fails. For a non-pointer `bool`, there's unfortunately no way to distinguish "not provided" from "explicitly false" without using a pointer.

This is a known pain point and we're tracking it. Thanks for reporting.
